File: crm/frontend/src/types/lead.ts
export type LeadStatus =
  | 'new'
  | 'researching'
  | 'contacted'
  | 'meeting_scheduled'
  | 'proposal_sent'
  | 'negotiating'
  | 'won'
  | 'lost';

export type LeadSource =
  | 'scraped_permit'
  | 'scraped_news'
  | 'scraped_planning'
  | 'manual'
  | 'referral';

export interface Lead {
  id: string;
  project_name: string;
  status: LeadStatus;
  source: LeadSource;
  priority: number;
  unit_count: number | null;
  estimated_value: string | null;
  project_type: string | null;
  construction_phase: string | null;
  estimated_completion: string | null;
  address: string | null;
  city: string | null;
  state: string | null;
  zip_code: string | null;
  next_action: string | null;
  next_action_date: string | null;
  loss_reason: string | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
  contact_count: number;
  activity_count: number;
}

export interface PipelineColumn {
  status: LeadStatus;
  count: number;
  leads: Lead[];
}

export const STATUS_LABELS: Record<LeadStatus, string> = {
  new: 'New Lead',
  researching: 'Researching',
  contacted: 'Contacted',
  meeting_scheduled: 'Meeting Scheduled',
  proposal_sent: 'Proposal Sent',
  negotiating: 'Negotiating',
  won: 'Won',
  lost: 'Lost',
};

export const STATUS_COLORS: Record<LeadStatus, string> = {
  new: 'bg-blue-100 text-blue-800',
  researching: 'bg-purple-100 text-purple-800',
  contacted: 'bg-yellow-100 text-yellow-800',
  meeting_scheduled: 'bg-orange-100 text-orange-800',
  proposal_sent: 'bg-indigo-100 text-indigo-800',
  negotiating: 'bg-pink-100 text-pink-800',
  won: 'bg-green-100 text-green-800',
  lost: 'bg-red-100 text-red-800',
};

File: crm/backend/app/schemas/lead.py
from typing import Optional
from pydantic import BaseModel


class LeadCreate(BaseModel):
    project_name: str
    status: str = "new"
    source: str = "manual"
    priority: int = 3
    unit_count: Optional[int] = None
    estimated_value: Optional[str] = None
    project_type: Optional[str] = None
    construction_phase: Optional[str] = None
    estimated_completion: Optional[str] = None
    address: Optional[str] = None
    city: Optional[str] = None
    state: Optional[str] = None
    zip_code: Optional[str] = None
    next_action: Optional[str] = None
    next_action_date: Optional[str] = None
    notes: Optional[str] = None


class LeadUpdate(BaseModel):
    project_name: Optional[str] = None
    status: Optional[str] = None
    source: Optional[str] = None
    priority: Optional[int] = None
    unit_count: Optional[int] = None
    estimated_value: Optional[str] = None
    project_type: Optional[str] = None
    construction_phase: Optional[str] = None
    estimated_completion: Optional[str] = None
    address: Optional[str] = None
    city: Optional[str] = None
    state: Optional[str] = None
    zip_code: Optional[str] = None
    next_action: Optional[str] = None
    next_action_date: Optional[str] = None
    loss_reason: Optional[str] = None
    notes: Optional[str] = None


class LeadResponse(BaseModel):
    id: str
    project_name: str
    status: str
    source: str
    priority: int
    unit_count: Optional[int] = None
    estimated_value: Optional[str] = None
    project_type: Optional[str] = None
    construction_phase: Optional[str] = None
    estimated_completion: Optional[str] = None
    address: Optional[str] = None
    city: Optional[str] = None
    state: Optional[str] = None
    zip_code: Optional[str] = None
    next_action: Optional[str] = None
    next_action_date: Optional[str] = None
    loss_reason: Optional[str] = None
    notes: Optional[str] = None
    created_at: str
    updated_at: str
    contact_count: int = 0
    activity_count: int = 0

    class Config:
        from_attributes = True


class PipelineResponse(BaseModel):
    status: str
    count: int
    leads: list[LeadResponse]
